dts/bindings/charger/maxim,max20335-charger.yaml

```diff
@@ -8,11 +8,70 @@ include: battery.yaml
 compatible: "maxim,max20335-charger"
 
 properties:
-  constant-charge-current-max-microamp:
-    required: true
-
   constant-charge-voltage-max-microvolt:
     required: true
+    enum:
+      - 4050000
+      - 4100000
+      - 4150000
+      - 4200000
+      - 4250000
+      - 4300000
+      - 4350000
+      - 4400000
+      - 4450000
+      - 4500000
+      - 4550000
+      - 4600000
+
+  chgin-to-sys-current-limit-microamp:
+    type: int
+    enum:
+      - 0
+      - 100000
+      - 500000
+      - 1000000
+    description: |
+      CHGIN to SYS path current limitter configuration.
+      Refer to ILimCntl register description for details.
+
+  system-voltage-min-threshold-microvolt:
+    type: int
+    enum:
+      - 3600000
+      - 3700000
+      - 3800000
+      - 3900000
+      - 4000000
+      - 4100000
+      - 4200000
+      - 4300000
+    description: |
+      System voltage minimum threshold. When SYS path
+      voltage drops to this level, the charger current
+      is reduced to prevent battery damage.
+
+  recharge-threshold-microvolt:
```

Comment:
https://www.kernel.org/doc/Documentation/devicetree/bindings/power/supply/battery.yaml

We may want to have this property name match Linux as `re-charge-voltage-microvolt` and also add it to `zephyr/dts/bindings/battery/battery.yaml`

Reply:
It seems it requires a note in migration-guide.

Reply:
@LukaszMadejGrinn, if we resolve this, I am ready to approve

Reply:
@rriveramcrus Its done. Please have a look. I have also added some notes to release notes and migration guide but I'm not sure if I've done this correctly.